### PR TITLE
Readme cleanup

### PR DESCRIPTION
**Pull request by smarlowucf**
### What does this PR do? Why are we making this change?

- Cleanup old diagrams and json examples which no longer match the current architecture
- Cleanup existing readme
- Consolidate mash docs into readme
- Remove old requirements on lxml and urllib3

### CODE REVIEW

**Review by jesusbv — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,104 +1,513 @@
...
+  --zone {gce_zone} \
+  --bucket {storage_bucket_name} \
+  --credentials /path/to/service/account/file.json
+```
+
+Publishing projects in GCE do not have permissions to create new instances
+for testing. Therefore for any publishing projects a set of testing
+credentials is required for launching and testing images. To add a publishing
+project with a testing account configured there are two more arguments in
+command line:
```
Comment (rjschwei) [inline README.md:339]: s/in command/in the command/

File: README.md [hunk 2]
```diff
@@ -1,104 +1,513 @@
...
-- werkzeug
-- google-auth
-- google-cloud-storage
-- google-api-python-client
-- aliyun-img-utils>=1.3.0
+See [.virtualenv.requirements.txt](https://github.com/SUSE-Enceladus/mash/.virtualenv.requirements.txt)
+
+## Service Design
+
+The implementation of the Mash is based on a collection of services. Each service completes
```
Comment (rjschwei) [inline README.md:427]: s/the Mash/Mash/

File: README.md [hunk 3]
```diff
@@ -1,104 +1,513 @@
...
+an individual task of the process. So far the following services exist:
```
Comment (rjschwei) [inline README.md:428]: s/So far/At present/

File: README.md [hunk 4]
```diff
@@ -1,104 +1,513 @@
...
+
+- __OBS (retrieval) service__:
+  
+  Images are built in a build service. The retrieval service currently relies
+  on [OBS](https://openbuildservice.org/). The service Validates a compute
```
Comment (rjschwei) [inline README.md:433]: s/Validates a/validates that a/

File: README.md [hunk 5]
```diff
@@ -1,104 +1,513 @@
...
+
+  Duplicates or copies the newly created compute image to all 
+  available regions.
+
+- __Publish service__:
+
+  Makes a compute image publically available.
+
+  __NOTE__:
+   the publish implementation does not fully cover all registration
```
Comment (rjschwei) [inline README.md:472]: s/registration/registrations/

File: README.md [hunk 6]
```diff
@@ -1,104 +1,513 @@
...
+   into the cloud service provider marketplace.
```
Comment (rjschwei) [inline README.md:473]: s/marketplace/marketplaces/

**Review by rjschwei — CHANGES_REQUESTED** [review]

File: README.md [hunk 7]
```diff
@@ -1,104 +1,513 @@
...
+The following command will generate the necessary json service principal file:
+
+```bash
+$ az ad sp create-for-rbac \
+  --sdk-auth \
+  --role Contributor \
+  --scopes /subscriptions/{subscription_id} \
+  --name "{name}" > /path/to/credentials.json
+```
+
```
[the remark this answers is not in the document]
Comment (smarlowucf) [reply README.md:209]: What do you mean? The --sdk-auth being deprecated? az cli can be authorized using a service principal the same as mash is using.

[the remark this answers is not in the document]
Comment (rjschwei) [reply README.md:209]: Maybe something like this. 

"""If you also use the Azure CLI tools (az) do not confuse the credentials. The az toolset uses a different set of credentials which will not work with mash"""

[the remark this answers is not in the document]
Comment (smarlowucf) [reply README.md:209]: I don't see the advantage to such a message. Mash uses service principles for Azure authentication. You can login into Azure CLI with the same service principle json file. Thus stating the credentials are different to me only adds confusion and is really an invalid statement.

**Review by rjschwei — APPROVED** [review]

File: README.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (smarlowucf) [reply README.md:209]: Again if we are talking about the deprecated --sdk-auth option for service principles, the story there has not been decided upstream. Thus I don't think it's useful for us to make any statement in that regard prematurely.

**Pull request merged**